drivers/storage/portworx/portworx.go

```diff
@@ -235,6 +235,12 @@ func (p *portworx) SetDefaultsOnStorageCluster(toUpdate *corev1.StorageCluster)
 		}
 
 		if pxutil.IsCSIEnabled(toUpdate) {
+			// Enable CSI snapshot controller by default if it's not configured on k8s 1.17+
+			if toUpdate.Spec.CSI.InstallSnapshotController == nil &&
+				p.k8sVersion != nil && p.k8sVersion.GreaterThanOrEqual(k8sutil.K8sVer1_17) {
+				toUpdate.Spec.CSI.InstallSnapshotController = boolPtr(true)
+			}
+
```

Comment:
Let's move this code to https://github.com/libopenstorage/operator/blob/4ad3571c9d6d93ac68c5dcb567a101eb01407ffb/drivers/storage/portworx/portworx.go#L721
This is not the right place to set the CSI default parameters.

Reply:
@piyush-nimbalkar we need k8s version to set it only for k8s 1.17+, but I don't see anything else in SetPortworxDefaults is k8s version related, so I put it here, but I could pass the k8s version to `SetPortworxDefaults` as well. wdyt?

Reply:
so there's another function I can call in SetPortworxDefaults is `GetVersion()` which uses `coreops.Instance()`, so this will work. I agree it's better to move the function and I originally did this, and it shouldn't hurt to call the GetVersion function, I'll change it.

Reply:
I chose to pass the k8s version to SetPortworxDefaults as the final solution.